### ISSUE TITLE
es7 binding operator doesn't work

### ISSUE THREAD

**Issue by shevchuk** [mooz/js2-mode#427]
There is a new bind syntax introduced in ES7:

```javascript
// Binding a function to a context
let log = ::console.log;
```

Which currently generates error:

```javascript
line 2: syntax error
line 2: missing ; before statement
```

**Comment by fxfactorial** [comment]
that isn't JS though, its a babel plugin and hopefully never becomes real JS.

**Comment by carlosyslas** [comment]
I agree it should not be supported officially by `js2-mode` but some of us have to already existing code that uses that syntax. Is there any easy hack for at least prevent the syntax highlighting to get broken?

**Comment by dgutov** [comment]
I don't know how we'd do that. The parser here is not made to be especially forgiving.

**Comment by gomain** [comment]
This (and other @babel features) will become a no-go for adopting js2-mode. There needs to be a mechanism in altering how the parser works.

**Comment by dgutov** [comment]
If somebody wants to try their hand at adding an extensibility framework, please be my guest.

Otherwise, `js-mode` is still an option when you have to work with more complex syntaxes, though it's not as feature-rich.

**Comment by shevchuk** [comment]
not fully related to this topic but still worth to mention, the js2-mode is also not working if flow types are used in the file. So I think all these may have a common solution with this extensibility framework that Dmitry mentioned
